lib/min_heap.rb

```diff
@@ -14,18 +14,29 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: ? O(logn)
+  # Space Complexity: ? O(1)
   def add(key, value = key)
-    raise NotImplementedError, "Method not implemented yet..."
+    #pushing key value of node into the store array
+    @store << HeapNode.new(key, value)
+    # add
+    heap_up(@store.length - 1)
   end
 
   # This method removes and returns an element from the heap
   #   maintaining the heap structure
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(logn)
+  # Space Complexity: O(1)
   def remove()
```

Comment:
👍  However since you are doing recursion the space complexity is O(log n) due to the call stack.